rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/Tile.java

```diff
@@ -0,0 +1,357 @@
+            }
+        });
+    }
+
+    private void addMouseOutHandler() {
+        this.addMouseOutHandler(event -> {
+            dirty = true;
+            hover.hide();
+            CompactView.setGlobalHover(null);
+        });
+    }
+
+    public void refresh(Node n) {
+        this.node = n;
+        this.setUrl(n.getIcon());
+        this.setHoverNodeLabel();
+    }
+
+    public void refresh(NodeSource ns) {
+        this.nodesource = ns;
+        this.setUrl(ns.getIcon());
+    }
+
+    private void setHoverNodeLabel() {
+        String str = "Provider: " + node.getNodeProvider() + "<br>";
+        if (node.getNodeOwner().trim().length() > 0) {
+            str += "Used by: " + node.getNodeOwner() + "<br>";
+        }
+        str += node.getNodeState().toString() + " since " + JSUtil.getTime(node.getTimeStamp());
+        this.hoverLabel.setContents(str);
+    }
+
+    void setSelectedTile(boolean selected) {
+        if (selected) {
+            this.getElement().getStyle().setBackgroundColor("#7caaf7");
+            this.getElement().getStyle().setBorderColor("#d9e4f6");
+            this.highlightGroup(true);
+        } else {
+            this.getElement().getStyle().setBackgroundColor("white");
+            this.getElement().getStyle().setBorderColor("white");
+            this.highlightGroup(false);
+        }
+    }
+
+    private void highlightGroup(boolean selected) {
+        // highlight every node following this one in the compactPanel layout,
+        // until we hit one from another host/ns
+        if (host != null || nodesource != null) {
+            int id;
+            if (host == null) {
+                id = panel.indexOf(nodesource).get();
+            } else {
+                id = panel.indexOf(host).get();
+            }
+
+            while (true) {
+                id++;
+                if (panel.getWidgetCount() == id)
+                    break;
+
+                Tile tile = (Tile) panel.getWidget(id);
+                // reaching the end
+                if (tile == null) {
+                    break;
+                }
+                // reaching next host/ms
+                if ((this.host != null && tile.host != null) || tile.nodesource != null) {
+                    break;
+                }
+                if (selected) {
+                    tile.getElement().getStyle().setBackgroundColor("#d9e4f6");
+                    tile.getElement().getStyle().setBorderColor("#d9e4f6");
+                } else {
+                    tile.getElement().getStyle().setBackgroundColor("white");
+                    tile.getElement().getStyle().setBorderColor("white");
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        Tile t = (Tile) o;
+        if (this.node != null && t.node != null) {
+            return this.node.getNodeUrl().equals(t.node.getNodeUrl());
+        } else if (this.host != null && t.host != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "t" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)